clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -84,6 +85,9 @@ private enum SubscriptionType {
     /* the pattern user has requested */
     private Pattern subscribedPattern;
 
+    /* we should rename this to something more specific */
```

Comment:
Agree that it's confusing but can't think of a better naming. I would suggest though that we add a proper comment, stating that this represents the RE2J regex (vs the java regex represented by the `Pattern subscribedPattern`